src/Components/Project/ProjectModal/Project.tsx

```diff
@@ -1,37 +1,22 @@
+import { useHandleClickModalBtn } from "hook";
+import { ModalStateType } from "../../../Utils/GlobalTypes";
 import React, { useCallback } from "react";
-import ProjectModal from "./ProjectModal/ProjectModal";
-import FieldChoice from "./FieldChoice/FieldChoice";
-import { useModalContext } from "Utils/Contexts/ModalContext";
 
-interface ProjectProps {
-  state: string;
-}
-
-const Project = (props: ProjectProps) => {
-  const { addModal } = useModalContext();
+function Project(props: { state: ModalStateType }) {
+  const { state } = props;
 
-  const handleClickRegister = useCallback(() => {
-    addModal({
-      title: "",
-      element: <ProjectModal state={props.state} />,
-      showOnlyBody: true,
-      width: "1450px",
-      height: "1000px",
-    });
-  }, []);
+  const handleClickRegister = useHandleClickModalBtn({
+    modalName: "Project",
+    state,
+  });
 
-  const handleClickRegisterToChoice = useCallback(() => {
-    addModal({
-      title: "",
-      element: <FieldChoice state={props.state} />,
-      showOnlyBody: true,
-      width: "1450px",
-      height: "1000px",
-    });
-  }, []);
+  const handleClickRegisterToChoice = useHandleClickModalBtn({
+    modalName: "FieldChoice",
+    state,
+  });
 
   const openModal = () => {
-    switch (props.state) {
+    switch (state) {
```

Comment:
openModal state is only one, we needless switch
because we have `if` or `object`